core/unix/loader_macos.c

```diff
@@ -54,13 +77,67 @@ privload_mod_tls_init(privmod_t *mod)
 void *
 privload_tls_init(void *app_tls)
 {
+#if defined(AARCH64)
+    void **cur_tls = (void **)read_thread_register(TLS_REG_LIB);
+    if (cur_tls && !pthread_ptr_munge_token) {
+        pthread_ptr_munge_token = (uintptr_t) * (cur_tls + TSD_PTR_MUNGE);
+    }
+
     /* XXX i#1285: implement MacOS private loader */
+    byte *pthread = NULL;
+
+    /* We use the mach vm_allocate API here since client threads may need
+     * a valid TLS even after the heap has been cleaned up.
+     */
+    IF_DEBUG(kern_return_t res =)
+    vm_allocate(mach_task_self(), (vm_address_t *)&pthread, PAGE_SIZE,
+                true /* anywhere */);
+    ASSERT(res == KERN_SUCCESS);
+
+    memset(pthread, 0, PAGE_SIZE);
+
+    ASSERT(ALIGNED(pthread, PAGE_SIZE));
+
+    uint64_t *tls = (uint64_t *)(pthread + PTHREAD_TLS_OFFSET);
+    tls[TSD_MACH_THREAD_SELF] = mach_thread_self();
+    tls[TSD_ERRNO] = (uint64_t)(pthread + ERRNO_OFFSET);
+    tls[TSD_THREAD_SELF] = (uint64_t)(pthread);
+
+    /* pthread->_sig */
+    void *sig = pthread;
+    if (proc_has_feature(FEATURE_PAUTH)) { /* XXX: what is the equivalent of
+                                              __has_feature(ptrauth_calls)? */
+        uint64_t modifier = 0x5b9;
+        __asm__ volatile("pacdb %[ptr], %[mod]"
```

Comment:
Add a comment explaining what this does, for readers who have no idea what "pacdb" does.